src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -32,19 +32,28 @@
  * {@code String s = Integer.toString(i = 2);}.
  * </p>
  * <p>
- * Rationale: With the exception of {@code for} iterators and assignment in {@code while} idiom,
+ * Rationale: With the exception of loop idioms,
  * all assignments should occur in their own top-level statement to increase readability.
  * With inner assignments like the one given above, it is difficult to see all places
  * where a variable is set.
  * </p>
  * <p>
- * Note: Check allows usage of the popular assignment in {@code while} idiom:
+ * Note: Check allows usage of the popular assignment in loops:
  * </p>
  * <pre>
  * String line;
- * while ((line = bufferedReader.readLine()) != null) {
- *   // process the line
+ * while ((line = bufferedReader.readLine()) != null) { // OK
+ *     // process the line
  * }
+ *
+ * for (;!((line = file.read()) != null);) { // OK
```

Comment:
`!((line = file.read()) != null)`
No need for additional example complications, make simple like in while example

Reply:
Done. I just want to let you know that this example was taken from while examples, and I haven't removed that yet, ok.

Reply:
it remains the same

Reply:
done

Reply:
please take a look now